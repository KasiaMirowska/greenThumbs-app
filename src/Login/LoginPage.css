.auth{
    background-position: right bottom, left top; 
    background-size: cover; 
    background-size: 100%;
    /* background-repeat: no-repeat; */
     background-image: url(https://green-images.s3.amazonaws.com/Image+URL.jpg); 
     padding-bottom: 300px;
    padding-top: 30px;
   
}

#auth {
   background-color:rgb(253, 235, 212) ;
   padding-left: 30px;
   display: flex;
   max-height: 100px;
}
.auth-form {
    padding-left: 30px;
    padding: 40px;
   
}
.form__field3 {
    width: 30%;
    margin: 5px;
    background-color:rgb(230, 251, 245) ;
}