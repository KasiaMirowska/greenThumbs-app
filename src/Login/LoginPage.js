import React from 'react';
import LoginForm from './LoginForm';

export default class LoginPage extends React.Component {
    handleLogin = () => {
        const {location, history} = this.props;
        const destination = (location.state || {}).state || '/';
        history.push(destination);
    }

    render() {
        return (
            <div>
                <h1>LOGIN</h1>
                <LoginForm onLoginSuccess={this.handleLogin}/>
            </div>
        )
    };
};