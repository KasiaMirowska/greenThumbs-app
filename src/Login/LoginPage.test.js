import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter as BR} from 'react-router-dom';
import LoginPage from './LoginPage';


describe('List component', () => {
    const props = {
        location: {state: ''},
        history: {push: '/'}
    }
   
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><LoginPage {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
   
    it('renders UI as expected', () => {
        const item= renderer.create(<BR><LoginPage {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })
})