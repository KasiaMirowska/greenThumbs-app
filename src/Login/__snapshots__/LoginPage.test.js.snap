// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<div
  className="auth"
>
  <div
    id="auth"
  >
    <h1>
      LOGIN 
    </h1>
    <div
      className="error"
    />
  </div>
  <div>
    <form
      className="auth-form"
      onSubmit={[Function]}
    >
      <input
        className="form__field form__field3"
        id="username"
        onClick={[Function]}
        placeholder="username"
        required={true}
        type="input"
      />
      <br />
      <input
        className="form__field form__field3"
        id="password"
        onClick={[Function]}
        placeholder="password"
        required={true}
        type="password"
      />
      <button
        type="submit"
      >
        Submit
      </button>
      <button
        className="login-button"
        type="reset"
      >
        Cancel
      </button>
    </form>
  </div>
</div>
`;
